tests/unittests/server_embeddings_v2_tests.py

```
@@ -0,0 +1,626 @@
+        self.assertIsNone(res["media"])
+        self.assertTrue(res["filepath"].endswith("clip.mp4"))
+
+    @drop_datasets
+    def test_negative_slices_are_rejected(self):
+        # Python's negative slicing would "work" and return unrelated
+        # wire-order rows — a protocol violation must raise instead
+        dataset, _ = _make_samples_run()
+        base = {"datasetName": dataset.name, "brainKey": "viz"}
+
+        for bad in ({"offset": -1}, {"limit": -5}):
+            with self.assertRaises(ValueError):
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)